library/core/src/iter/adapters/fuse.rs

```diff
@@ -24,6 +24,10 @@ impl<I> Fuse<I> {
     pub(in crate::iter) fn new(iter: I) -> Fuse<I> {
         Fuse { iter: Some(iter) }
     }
+    /// Whether the iterator is fused (i.e. it will return `None` forever)
+    pub(in crate::iter) fn is_fused(&self) -> bool {
+        self.iter.is_none()
```

Comment:
This isn't going to work due to the FusedIterator optimization (see comment on `iter` above), I'm not sure there's an easy way to detect what you want via Fuse itself.

Reply:
I agree. We don't know whether the inner iterator is fused if itself is a `FusedIterator`.